jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java

```diff
@@ -48,10 +48,9 @@ public LocalDateArgument(final LocalDate value) {
         @Override
         public void apply(final int position, final PreparedStatement statement, final StatementContext ctx) throws SQLException {
             if (value != null) {
-                // ISO8601 format
-                statement.setString(position, value.toString());
+                statement.setDate(position, new java.sql.Date(value.toDate().getTime()));
```

Comment:
That makes me extremely nervous! We had a bug with the way we stored date in the DB which if remember correctly lead us to use that approach. I spent an entire week chasing the bug... why do we need the change?